### PR TITLE
Add documentation links for ceph-based alerts

### PR DESCRIPTION
**Pull request by rexagod**
Signed-off-by: Pranshu Srivastava <rexagod@gmail.com>

***

Will be merged once https://github.com/openshift/console/pull/12059/files is included in the dynamic SDK.

### CODE REVIEW

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx [hunk 1]
```diff
@@ -54,6 +54,7 @@ export const CephAlerts: React.FC = () => {
           <AlertItem
             key={alertURL(alert, alert?.rule?.id)}
             alert={alert as any}
+            documentationLink={alert?.annotations?.storage_type === "ceph" ? `https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.12/html-single/troubleshooting_openshift_data_foundation/index#${_.toLower(alert?.labels?.alertname)}_rhodf` : null}
```
Comment (bipuladh) [inline packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:57]: Decompose this to multiple functions. One to  generate the alert link other to verify its a ceph storage_type. And one more which takes those two functions and either passes documentationLink or `undefined`

**Review by bipuladh — CHANGES_REQUESTED** [review]
Need to perform similar check for Noobaa.

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx [hunk 2]
```diff
@@ -46,6 +49,22 @@ export const CephAlerts: React.FC = () => {
...
+    return `https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.12/html-single/troubleshooting_openshift_data_foundation/index#${_.toLower(alert?.labels?.alertname)}_rhodf`;
+  };
+
+  const isOdfManaged = useFlag(ODF_MANAGED_FLAG);
+  const getDocumentationLink = (alert: Alert): string => {
+  if (!isOdfManaged && isCephBasedAlert(alert)) {
+      return generateDocumentationLink(alert);
+    }
+    return null;
+  };
```
Comment (SanjalKatiyar) [inline packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:66]: let's move generateDocumentationLink and getDocumentationLink outside the FC.

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx (hunk 1 above)
Comment (rexagod) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx]: ACK.

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx (hunk 2 above)
Comment (rexagod) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx]: Done.

**Comment by rexagod** [comment]
@bipuladh Do you know when is the next dynamic SDK release queued for (as we want https://github.com/openshift/console/pull/12059/files to be there in the SDK for this merge to happen)? Should I ask Sam to accelerate the process, if possible, so that we can do this before freeze? Or, since this is only on our end now (console-side PR is in) can we wait for the release to happen without the need to push the console team?

**Comment by bipuladh** [comment]
@rexagod lets get this merged.

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx [hunk 3]
```diff
@@ -54,6 +73,8 @@ export const CephAlerts: React.FC = () => {
           <AlertItem
             key={alertURL(alert, alert?.rule?.id)}
             alert={alert as any}
+            // @ts-ignore
```
Comment (rexagod) [inline packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:76]: @bipuladh I think we can go ahead with the merge now.

Comment (rexagod) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:76]: cc @SanjalKatiyar

**Comment by SanjalKatiyar** [comment]
need to run "yarn format", looks like "lint-staged" working as intended...

File: packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx (hunk 3 above)
Comment (SanjalKatiyar) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:76]: just to be sure, did u test it once on latest OCP ?

Comment (rexagod) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:78]: Getting this on OCP 4.10.

<details>
<summary>Dashboard</summary>


image


</details>

Comment (rexagod) [reply packages/ocs/dashboards/persistent-internal/status-card/status-card.tsx:78]: On OCP 4.10 (tested with `console` on r-4.10 and master) on this PR, I'm getting this. Not sure why https://github.com/openshift/console/pull/11976 isn't being picked up as this is in `console#master` already, maybe something's disabling it? Last time I checked this feature, I used 4.12, so maybe it's probable that 4.10 could be blocking it?

<details>
<summary>Details</summary>

image

</details>

**Comment by rexagod** [comment]
Verified.

<details>
<summary>Details</summary>

image
image
Screenshot from 2022-11-11 16-34-05

</details>

**Pull request merged**